iree/test/e2e/matmul/generate_e2e_matmul_tests.py

```diff
@@ -572,7 +572,7 @@ def parse_arguments():
       "--module_path",
       type=str,
       help=
-      "Module path (typically .vmfb) to be referenced in the output trace. Should match the output path of the iree-translate command generating the module.",
+      "Module path (typically .vmfb) to be referenced in the output trace. Should match the output path of the iree-compile command generating the module.",
```

Comment:
Something else is hitting errors with Android cross compilation. I don't understand this part MLIR/codegen well enough to suggest what is wrong though.

`error: cannot be converted to LLVM IR: missing LLVMTranslationDialectInterface registration for dialect for op: arm_neon.intr.sdot`

https://source.cloud.google.com/results/invocations/c6e2a710-dd5b-47fd-bfa8-f09eee5e078e/targets/iree%2Fgcp_ubuntu%2Fcmake%2Fandroid%2Farm64-v8a%2Fpresubmit/log

```
[294/475] Generating e2e_matmul_mmt4d_i8_intrinsics_small_dylib-llvm-aot_dylib_dotprod.vmfb
FAILED: iree/test/e2e/matmul/e2e_matmul_mmt4d_i8_intrinsics_small_dylib-llvm-aot_dylib_dotprod.vmfb 
cd /tmpfs/src/github/iree/build-android/iree/test/e2e/matmul && /tmpfs/src/github/iree/build-android/../build-host/install/bin/iree-compile -iree-mlir-to-vm-bytecode-module -mlir-print-op-on-diagnostic=false --iree-hal-target-backends=dylib-llvm-aot --iree-codegen-mmt4d-use-intrinsics "--iree-flow-mmt4d-target-options=enable_generic_slow arch=aarch64 features=+dotprod" --iree-llvm-target-triple=aarch64-none-linux-android29 --iree-llvm-target-cpu-features=+dotprod /tmpfs/src/github/iree/build-android/iree/test/e2e/matmul/e2e_matmul_mmt4d_i8_intrinsics_small_dylib-llvm-aot_dylib_dotprod.mlir -o /tmpfs/src/github/iree/build-android/iree/test/e2e/matmul/e2e_matmul_mmt4d_i8_intrinsics_small_dylib-llvm-aot_dylib_dotprod.vmfb
/tmpfs/src/github/iree/build-android/iree/test/e2e/matmul/e2e_matmul_mmt4d_i8_intrinsics_small_dylib-llvm-aot_dylib_dotprod.mlir:2:13: error: cannot be converted to LLVM IR: missing `LLVMTranslationDialectInterface` registration for dialect for op: arm_neon.intr.sdot
  %result = linalg.matmul ins(%lhs, %rhs: tensor<?x?xi8>, tensor<?x?xi8>) outs(%acc: tensor<?x?xi32>) -> tensor<?x?xi32>
            ^
/tmpfs/src/github/iree/build-android/iree/test/e2e/matmul/e2e_matmul_mmt4d_i8_intrinsics_small_dylib-llvm-aot_dylib_dotprod.mlir:1:1: note: called from
func @matmul_DYNxDYNxi8_times_DYNxDYNxi8_into_DYNxDYNxi32(%lhs: tensor<?x?xi8>, %rhs: tensor<?x?xi8>, %acc: tensor<?x?xi32>) -> tensor<?x?xi32> {
^
/tmpfs/src/github/iree/build-android/iree/test/e2e/matmul/e2e_matmul_mmt4d_i8_intrinsics_small_dylib-llvm-aot_dylib_dotprod.mlir:0:0: error: failed to translate the MLIR LLVM dialect to the native llvm::Module
/tmpfs/src/github/iree/build-android/iree/test/e2e/matmul/e2e_matmul_mmt4d_i8_intrinsics_small_dylib-llvm-aot_dylib_dotprod.mlir:0:0: error: failed to serialize executable for target backend llvm
/tmpfs/src/github/iree/build-android/iree/test/e2e/matmul/e2e_matmul_mmt4d_i8_intrinsics_small_dylib-llvm-aot_dylib_dotprod.mlir:0:0: error: failed to serialize executables
compilation from source to vm failed
[295/475] Generating check_dylib-llvm-aot-host_dylib_constant.mlir_module.vmfb
'-avx512pf' is not a recognized feature for this target (ignoring feature)
'-tsxldtrk' is not a recognized feature for this target (ignoring feature)
```

Reply:
Oh I think I know what is involved there. We really need to encapsulate this all somewhere...

Reply:
Ok, cleaned this up quite a bit. CI should pass now, give or take Bazel.